runtime/server/server.go

```diff
@@ -87,7 +89,20 @@ func (s *Server) ServeHTTP(ctx context.Context) error {
 	return graceful.ServeHTTP(ctx, server, s.opts.HTTPPort)
 }
 
-// HTTP handler serving REST gateway
+// CustomErrorToCode returns the Code of the error if it is a Status error
+// otherwise use status.FromContextError to determine the Code.
+// Log level for error codes is defined in logging.DefaultServerCodeToLevel
+func CustomErrorToCode(err error) codes.Code {
+	if se, ok := err.(interface {
+		GRPCStatus() *status.Status
+	}); ok {
+		return se.GRPCStatus().Code()
+	}
+	contextStatus := status.FromContextError(err)
+	return contextStatus.Code()
+}
```

Comment:
I think its ok for now since part from `codes.Internal`
```
codes.Unknown, codes.Unimplemented, codes.DataLoss
```
gets logged as error which seems valid, may be we can wait and change later if needed. Anyways no strong opinions here.

Reply:
With the new queries refactor, there will probably be more "unknown" errors returned, so would be nice to make those "info" messages instead